### PR TITLE
Add random suffix to testing namespace to avoid duplicate creation

### PR DESCRIPTION
**Pull request by pengyu-hou**
## Summary

There are some errors related to db already created error. Adding a random suffix to mitigate the issue. 

## Why / Goal

Improve flaky test

## Test Plan

- [ ] Added Unit Tests
- [x] Covered by existing CI
- [ ] Integration tested

## Checklist
- [ ] Documentation update

## Reviewers
@pkundurthy @hzding621

### CODE REVIEW

**Review by hzding621 — APPROVED** [review]
Can we also update `JoinFlowTest.scala`? Thanks!

**Review by pkundurthy — APPROVED** [review]
LGTM

File: spark/src/test/scala/ai/chronon/spark/test/JoinTest.scala [hunk 1]
```diff
@@ -61,7 +62,7 @@ class JoinTest {
   private val yearAgo = tableUtils.partitionSpec.minus(today, new Window(365, TimeUnit.DAYS))
   private val dayAndMonthBefore = tableUtils.partitionSpec.before(monthAgo)
 
-  private val namespace = "test_namespace_jointest"
+  private val namespace = "test_namespace_jointest" + "_" + Random.alphanumeric.take(6).mkString
```
Comment (pkundurthy) [inline spark/src/test/scala/ai/chronon/spark/test/JoinTest.scala:65]: @pengyu-hou - I think there are a couple of places where this is hard-coded so we might need to update these as well , or pass the random string into it.

Comment (pkundurthy) [reply spark/src/test/scala/ai/chronon/spark/test/JoinTest.scala:65]: https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/JoinTest.scala?L1128-1137

so maybe 
val random_suffix = Random.alphanumeric.take(6).mkString

and then use that in the references

File: spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala [hunk 2]
```diff
@@ -30,13 +30,13 @@ import org.junit.Assert._
...
 import scala.collection.mutable
-import scala.util.Try
+import scala.util.{Random, Try}
 
 class JoinUtilsTest {
 
   lazy val spark: SparkSession = SparkSessionBuilder.build("JoinUtilsTest", local = true)
   private val tableUtils = TableUtils(spark)
-  private val namespace = "joinUtil"
+  private val namespace = "joinUtil"  + "_" + Random.alphanumeric.take(6).mkString
```
Comment (pkundurthy) [inline spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala:39]: similar to above comment
hard-coded references might need to get the random string as well. 

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala?L239-240

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala?L277-279

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala?L334

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/JoinUtilsTest.scala?L351

File: spark/src/test/scala/ai/chronon/spark/test/LabelJoinTest.scala [hunk 3]
```diff
@@ -23,12 +23,14 @@ import org.apache.spark.sql.{Row, SparkSession}
...
 
+import scala.util.Random
+
 class LabelJoinTest {
   @transient private lazy val logger = LoggerFactory.getLogger(getClass)
 
   val spark: SparkSession = SparkSessionBuilder.build("LabelJoinTest", local = true)
 
-  private val namespace = "label_join"
+  private val namespace = "label_join" + "_" + Random.alphanumeric.take(6).mkString
```
Comment (pkundurthy) [inline spark/src/test/scala/ai/chronon/spark/test/LabelJoinTest.scala:33]: might need to update hard coded references

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/LabelJoinTest.scala?L59

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/LabelJoinTest.scala?L104-111

https://sourcegraph.d.musta.ch/github.com/airbnb/chronon/-/blob/spark/src/test/scala/ai/chronon/spark/test/LabelJoinTest.scala?L355-356
